lib/pbench/test/unit/server/query_apis/test_datasets_delete.py

```diff
@@ -160,6 +160,7 @@ def test_partial(
         monkeypatch,
         pbench_token,
         server_config,
+        current_user_drb,
```

Comment:
The `pbench_token` provides the authentication token used here. I don't immediately see the purpose of `current_user_drb`: it fakes out the `Auth.token_auth` call, but that would already be satisfied by the authorization header computed here from `pbench_token`. We shouldn't be doing both...

Reply:
Yeah, this test is getting weird. I am getting sqlalchemy `DetachedInstanceError` on the user instance for some reason. There is a user instance left around without any session attached to it resulting in the above error. I am guessing this is because of the `get_document_map` fixture that calls the `attach_dataset` fixture which creates the users but our code closes the session as soon as we encounter the APIAbort call. And this sort of creates this race condition where we are closing the session before the objects are refreshed for the unit test. However, I am failing to understand why it doesn't result in the same error when I use `pbench_token` and `current_user_drb` together, both are using `create_drb_user` fixture which creates the user. Another weird thing is that using `with client` to retain the DB session, it succeed locally when I run `pytest test_dataset_delete.py` but in Jenkins it fails.